lib/src/main/resources/com/diffplug/spotless/npm/prettier-serve.js

```diff
@@ -0,0 +1,93 @@
+const GracefulShutdownManager = require("@moebius/http-graceful-shutdown").GracefulShutdownManager;
+const express = require("express");
+const app = express();
+
+app.use(express.json({ limit: "50mb" }));
+const prettier = require("prettier");
+
+const fs = require("fs");
+
+var listener = app.listen(0, "127.0.0.1", () => {
```

Comment:
@JLLeitschuh does this ring any security bells for you?  We used to use J2V8 (basically node compiled for JVM interop), but it has fallen into disrepair.  @simschla has made an excellent little express-js server which runs in npm, and accepts requests from spotless as a bridge to the world of javascript formatting.

Here's how it works (correct if I'm wrong @simschla)

- js: open any available port in 127.0.0.1
- js: write that port and address to a temp file
- java: read that file
- java: send post requests to that address, and eventually post a `/shutdown` command
- it all closes

Reply:
perfectly summarized :)